packages/paste-core/components/form/src/FormInput.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import css from '@styled-system/css';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+
+export type InputTypes = 'text' | 'email' | 'hidden' | 'number' | 'password' | 'search' | 'tel';
+
+export interface FormInputProps extends React.InputHTMLAttributes<''> {
+  className?: never;
+  disabled?: boolean;
+  hasError?: boolean;
+  id: string;
+  name?: string;
+  placeholder?: string;
+  readOnly?: boolean;
+  required?: boolean;
+  style?: never;
+  type: InputTypes;
+  value?: string;
+  size?: never;
+  height?: never;
+  width?: never;
+}
+
+// eslint-disable-next-line emotion/syntax-preference
+const StyledFormInput = styled(Box)<FormInputProps>(({hasError}: FormInputProps) =>
+  css({
+    appearance: 'none',
+    outline: 'none',
+    resize: 'none',
+    width: '100%',
+    backgroundColor: 'colorBackgroundBody',
+    borderWidth: 'borderWidth10',
+    borderStyle: 'solid',
+    borderColor: hasError ? 'colorBorderError' : 'colorBorderInput',
+    borderRadius: 'borderRadius20',
+    paddingTop: 'space30',
+    paddingRight: 'space30',
+    paddingBottom: 'space30',
+    paddingLeft: 'space40',
+    fontSize: 'fontSize30',
+    lineHeight: 'lineHeight30',
+    fontWeight: 'fontWeightNormal',
+    color: 'colorText',
+    transition: 'border-color 100ms ease-in',
+
+    '&::placeholder': {
+      color: 'colorTextWeak',
+      fontStyle: 'italic',
+    },
+
+    '&:hover': {
+      borderColor: 'colorBorderPrimaryDarker',
+    },
+
+    '&:focus': {
+      borderColor: 'colorBorderPrimaryDarker',
+      boxShadow: 'shadowFocus',
+    },
+
+    '&:disabled': {
+      cursor: 'not-allowed',
+      backgroundColor: 'colorBackground',
+      borderColor: 'colorBorderLight',
+      color: 'colorTextWeak',
+    },
+
+    '&:read-only': {
+      cursor: 'not-allowed',
+      backgroundColor: 'colorBackground',
+      borderColor: 'colorBorderLight',
+      color: 'colorTextWeak',
+    },
+  })
+);
+
+const FormInput: React.FC<FormInputProps> = ({
+  disabled,
+  id,
+  hasError,
+  name,
+  placeholder,
+  readOnly,
+  required,
+  type,
+  value,
+  ...props
+}) => (
+  <StyledFormInput
+    {...safelySpreadBoxProps(props)}
+    aria-invalid={hasError}
+    aria-readonly={readOnly}
+    as="input"
+    disabled={disabled}
+    hasError={hasError}
+    id={id}
+    name={name}
+    placeholder={placeholder}
+    readOnly={readOnly}
+    required={required}
+    type={type}
+    value={value}
+  />
+);
```

Comment:
With this design, how are you going to handle the prefix and suffix children?

Reply:
Good question. I'm open to ideas on this. My initial thought is something like this:
```
<FormInputGroup>
    <FormInputAddon />
    <FormInput variant="addon" />
</FormInputGroup>
```

Reply:
So would the `FormInputGroup` have the border? That way you don't need to start conditionally setting border radius on things, which might not treat us too well in the future if we ever need to change or add anything.

Reply:
Mainly for reused purposes but also alignment. Textare and select all share the same border and prefix / suffix treatments, but select also has an addition icons.

![image](https://user-images.githubusercontent.com/368249/76133990-0b8f5300-5fd0-11ea-845f-ff7243bea02c.png)

![image](https://user-images.githubusercontent.com/368249/76134003-22ce4080-5fd0-11ea-8d5b-30b41259eae6.png)